src/main/java/org/apache/sling/distribution/journal/kafka/KafkaClientProvider.java

```diff
@@ -203,6 +222,14 @@ private void closeQuietly(final Closeable closeable) {
     @Nonnull
     private synchronized KafkaProducer<String, byte[]> buildKafkaProducer() {
         if (rawProducer == null) {
+            ClassLoader oldClassloader = Thread.currentThread().getContextClassLoader();
```

Comment:
What is the motivation for the class loader swap ?

Reply:
Kafka uses the wrong classloader by default in OSGi when loading the sasl classes.

Reply:
oh, thanks for the clarification.